include/rocksdb/advanced_options.h

```diff
@@ -1090,6 +1090,13 @@ struct AdvancedColumnFamilyOptions {
   // Dynamically changeable through the SetOptions() API.
   uint32_t bottommost_file_compaction_delay = 0;
 
+  // Enables additional integrity checks during reads/scans.
+  // Specifically, for skiplist-based memtables, we verify that keys visited
+  // are in order. This is helpful to detect corrupted memtable keys during
+  // reads. Enabling this feature incurs a performance overhead due to an
+  // additional key comparison during memtable lookup.
+  bool integrity_checks = false;
```

Comment:
This name is too general. We already have `paranoid_file_checks` and `force_consistency_checks` along with all the checksum options. I would suggest something like `paranoid_memory_checks` or `extra_read_time_checks`.

Reply:
Thanks for the suggestions, updated the option to `paranoid_memory_checks`.